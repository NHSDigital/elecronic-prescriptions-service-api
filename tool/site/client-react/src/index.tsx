import {Bundle} from "fhir/r4"
import * as React from "react"
import {
  createPrescriptionSummaryProps,
  PrescriptionSummary
} from "./components/prescription-summary/prescriptionSummary"
import {PageContainer} from "./components/pageContainer"
import ReactDOM = require("react-dom")

(async function() {
  const baseUrl = "/"
  const urlParams = new URLSearchParams(window.location.search)
  const prescriptionId = urlParams.get("prescription_id")
  const getPrescriptionResponse: Response = await fetch(`${baseUrl}prescription/${prescriptionId}`)
  const bundle: Bundle = await getPrescriptionResponse.json()
  const prescriptionSummaryProps = createPrescriptionSummaryProps(bundle)
  const content = (
    <PageContainer>
      <PrescriptionSummary
        patient={prescriptionSummaryProps.patient}
        practitionerRole={prescriptionSummaryProps.practitionerRole}
      />
    </PageContainer>
  )
  ReactDOM.render(content, document.getElementById("root"))
}())
