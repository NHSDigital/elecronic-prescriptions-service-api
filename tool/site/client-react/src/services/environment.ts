export type Environment = "internal-dev" | "int"

export const internalDev = "internal-dev"
export const int = "int"

export function isDev(environment: Environment): boolean {
  return environment === internalDev
}

export function isInt(environment: Environment): boolean {
  return environment === int
}
