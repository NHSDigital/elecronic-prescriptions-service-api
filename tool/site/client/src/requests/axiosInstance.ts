import axios from "axios"

export const axiosInstance = axios.create({
  validateStatus: () => true
})

// todo: refresh token if required
axiosInstance.interceptors.request.use(function (config) {
  return config
}, function (error) {
  return Promise.reject(error)
})
