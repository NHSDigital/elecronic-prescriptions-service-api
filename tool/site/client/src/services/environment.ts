export type Environment = "internal-dev" | "internal-dev-sandbox" | "internal-qa" | "int" | "sandbox"

export const internalDev = "internal-dev"
export const internalDevSandbox = "internal-dev-sandbox"
export const internalQa = "internal-qa"
export const int = "int"
export const sandbox = "sandbox"

export function isDev(environment: Environment): boolean {
  return environment === internalDev || environment === internalDevSandbox
}

export function isQa(environment: Environment): boolean {
  return environment === internalQa
}

export function isInt(environment: Environment): boolean {
  return environment === int
}

export function isSandbox(environment: Environment): boolean {
  return environment === sandbox
}
