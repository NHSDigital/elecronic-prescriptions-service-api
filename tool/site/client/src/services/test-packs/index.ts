import * as XLSX from "xlsx"
import * as fhir from "fhir/r4"
import {getMedicationRequestResources, getMessageHeaderResources} from "../../fhir/bundleResourceFinder"
import {Dispatch, SetStateAction} from "react"
import {createPatients, getPatient} from "./patients"
import {createPractitioners, getPractitioner} from "./practitioners"
import {
  getRowsFromSheet,
  parseOrganisationRowsOrDefault,
  parseParentOrganisationRowsOrDefault,
  parsePatientRowsOrDefault,
  parsePrescriptionRows,
  PrescriptionRow
} from "./xls"
import {createPractitionerRole} from "./practitionerRoles"
import {createCommunicationRequest} from "./communicationRequests"
import {createMessageHeader} from "./messageHeader"
import {createPlaceResources} from "./placeResources"
import {createMedicationRequests} from "./medicationRequests"
import {groupBy} from "./helpers"

export const createPrescriptionsFromExcelFile = (
  file: Blob,
  setPrescriptionsInTestPack: Dispatch<SetStateAction<any[]>>,
  setLoadPageErrors: Dispatch<SetStateAction<any>>
): void => {
  const reader = new FileReader()

  reader.onload = function (e) {
    const data = e.target.result
    const workbook = XLSX.read(data, {
      type: "binary"
    })

    const prescriptionRows = parsePrescriptionRows(getRowsFromSheet("Prescriptions", workbook), setLoadPageErrors)
    const patientRows = parsePatientRowsOrDefault(getRowsFromSheet("Patients", workbook, false), prescriptionRows.length)
    const prescriberRows = getRowsFromSheet("Prescribers", workbook, false)
    const organisationRows = parseOrganisationRowsOrDefault(getRowsFromSheet("Organisation", workbook, false), prescriptionRows.length)
    const parentOrganisationRows = parseParentOrganisationRowsOrDefault(getRowsFromSheet("Parent Organisation", workbook, false), prescriptionRows.length)
    const patients = createPatients(patientRows)
    const prescribers = createPractitioners(prescriberRows)
    const prescriptionType = getPrescriptionTypeFromRow(prescriptionRows)
    const places = createPlaceResources(prescriptionType, organisationRows, parentOrganisationRows)
    setPrescriptionsInTestPack(createPrescriptions(prescriptionType, patients, prescribers, places, prescriptionRows, setLoadPageErrors))
  }

  reader.onerror = function (ex) {
    console.log(ex)
  }

  reader.readAsBinaryString(file)
}

function createPrescriptions(
  prescriptionType: PrescriptionType,
  patients: Array<fhir.BundleEntry>,
  prescribers: Array<fhir.BundleEntry>,
  places: Array<fhir.BundleEntry>,
  rows: Array<PrescriptionRow>,
  setLoadPageErrors: Dispatch<SetStateAction<any>>
): any[] {
  const prescriptions = []
  const tests = groupBy(rows, (row: PrescriptionRow) => row.testId)

  tests.forEach(medicationItemRows => {
    const prescriptionRow = medicationItemRows[0]
    const patient = getPatient(patients, prescriptionRow)
    const prescriber = getPractitioner(prescribers, prescriptionRow)
    const nominatedPharmacy = prescriptionRow.nominatedPharmacy

    const prescriptionTreatmentTypeCode = getPrescriptionTreatmentType(prescriptionRow, setLoadPageErrors)

    switch (prescriptionTreatmentTypeCode) {
      case "acute":
        createAcutePrescription(prescriptionType, patient, prescriber, places, medicationItemRows, nominatedPharmacy, prescriptions)
        break
      case "continuous":
        createRepeatPrescribingPrescriptions(prescriptionType, patient, prescriber, places, medicationItemRows, nominatedPharmacy, prescriptions)
        break
      case "continuous-repeat-dispensing":
        createRepeatDispensingPrescription(prescriptionType, patient, prescriber, places, medicationItemRows, nominatedPharmacy, prescriptions)
        break
      default:
        throw new Error(`Invalid 'Prescription Treatment Type', must be one of: ${validFhirPrescriptionTypes.join(", ")}`)
    }
  })
  return prescriptions
}

const validFhirPrescriptionTypes = ["acute", "repeat-prescribing", "repeat-dispensing"]

function createAcutePrescription(
  prescriptionType: PrescriptionType,
  patient: fhir.BundleEntry,
  prescriber: fhir.BundleEntry,
  places: Array<fhir.BundleEntry>,
  prescriptionRows: PrescriptionRow[],
  nominatedPharmacy: string,
  prescriptions: any[]
) {
  const prescription = createPrescription(prescriptionType, patient, prescriber, places, prescriptionRows)

  updateNominatedPharmacy(prescription, nominatedPharmacy)
  prescriptions.push(JSON.stringify(prescription))
}

function createRepeatPrescribingPrescriptions(
  prescriptionType: PrescriptionType,
  patient: fhir.BundleEntry,
  prescriber: fhir.BundleEntry,
  places: Array<fhir.BundleEntry>,
  prescriptionRows: PrescriptionRow[],
  nominatedPharmacy: string,
  prescriptions: any[]
) {
  const prescriptionRow = prescriptionRows[0]
  const repeatsAllowed = prescriptionRow.repeatsAllowed
  for (let repeatsIssued = 0; repeatsIssued <= repeatsAllowed; repeatsIssued++) {
    const prescription = createPrescription(
      prescriptionType,
      patient,
      prescriber,
      places,
      prescriptionRows,
      repeatsIssued,
      repeatsAllowed
    )

    updateNominatedPharmacy(prescription, nominatedPharmacy)
    prescriptions.push(JSON.stringify(prescription))
  }
}

function createRepeatDispensingPrescription(
  prescriptionType: PrescriptionType,
  patient: fhir.BundleEntry,
  prescriber: fhir.BundleEntry,
  places: Array<fhir.BundleEntry>,
  prescriptionRows: Array<PrescriptionRow>,
  nominatedPharmacy: string,
  prescriptions: any[]
) {
  const prescriptionRow = prescriptionRows[0]
  const prescription = createPrescription(
    prescriptionType,
    patient,
    prescriber,
    places,
    prescriptionRows,
    0,
    prescriptionRow.repeatsAllowed
  )

  updateNominatedPharmacy(prescription, nominatedPharmacy)
  prescriptions.push(JSON.stringify(prescription))
}

function createPrescription(
  prescriptionType: PrescriptionType,
  patientEntry: fhir.BundleEntry,
  practitionerEntry: fhir.BundleEntry,
  places: Array<fhir.BundleEntry>,
  prescriptionRows: Array<PrescriptionRow>,
  repeatsIssued = 0,
  maxRepeatsAllowed = 0
): fhir.Bundle {

  const practitionerRoleEntry = createPractitionerRole(/*prescriptionType*/)

  if (prescriptionType === "trust-site-code") {
    (practitionerRoleEntry.resource as fhir.PractitionerRole).healthcareService = [
      {
        reference: "urn:uuid:54b0506d-49af-4245-9d40-d7d64902055e"
      }
    ]
  }

  const fhirPrescription: fhir.Bundle = {
    resourceType: "Bundle",
    id: prescriptionRows[0].testId,
    identifier: {
      system: "https://tools.ietf.org/html/rfc4122",
      value: "ea66ee9d-a981-432f-8c27-6907cbd99219"
    },
    type: "message",
    entry: [
      createMessageHeader(),
      patientEntry,
      practitionerEntry,
      practitionerRoleEntry,
      createCommunicationRequest(patientEntry, prescriptionRows[0].additionalInstructions)
    ]
  }

  createMedicationRequests(
    prescriptionRows,
    repeatsIssued,
    maxRepeatsAllowed
  ).forEach(medicationRequest =>
    fhirPrescription.entry.push(medicationRequest)
  )

  fhirPrescription.entry = [
    ...fhirPrescription.entry,
    ...places
  ]

  return fhirPrescription
}

export function getPrescriptionTreatmentType(row: PrescriptionRow, setLoadPageErrors?: Dispatch<SetStateAction<any>>): TreatmentType {
  const code = row.prescriptionTreatmentTypeCode
  if (!validFhirPrescriptionTypes.includes(code)) {
    // eslint-disable-next-line max-len
    const treatmentTypeInvalidError = `Treatment Type column contained an invalid value. 'Prescription Type' must be one of: ${validFhirPrescriptionTypes.join(", ")}`
    if (setLoadPageErrors) {
      setLoadPageErrors({details: [treatmentTypeInvalidError]})
    }
    throw new Error(treatmentTypeInvalidError)
  }
  switch (code) {
    case "acute":
      return "acute"
    case "repeat-prescribing":
      return "continuous"
    case "repeat-dispensing":
      return "continuous-repeat-dispensing"
  }
}

function getPrescriptionTypeFromRow(prescriptionRows: Array<PrescriptionRow>): PrescriptionType {
  const row = prescriptionRows[0]
  const prescriptionTypeCode = row.prescriptionTypeCode
  return getPrescriptionType(prescriptionTypeCode)
}

function updateNominatedPharmacy(bundle: fhir.Bundle, odsCode: string): void {
  if (!odsCode) {
    return
  }
  getMessageHeaderResources(bundle).forEach(messageHeader => {
    messageHeader.destination.forEach(destination => {
      destination.receiver.identifier.value = odsCode
    })
  })
  getMedicationRequestResources(bundle).forEach(function (medicationRequest) {
    medicationRequest.dispenseRequest.performer = {
      identifier: {
        system: "https://fhir.nhs.uk/Id/ods-organization-code",
        value: odsCode
      }
    }
  })
}

export type TreatmentType = "acute" | "continuous" | "continuous-repeat-dispensing"

export type PrescriptionType = "prescribing-cost-centre-0101" | "prescribing-cost-centre-non-0101" | "trust-site-code"

export function getPrescriptionType(prescriberType: string): PrescriptionType {
  if (prescriberType.startsWith("10")) {
    return "trust-site-code"
  }

  switch (prescriberType) {
    case "0101":
      return "prescribing-cost-centre-0101"
    case "0104":
    case "0105":
    case "0108":
    case "0125":
      return "prescribing-cost-centre-non-0101"
    default:
      throw new Error("Prescription type not handled")
  }
}
