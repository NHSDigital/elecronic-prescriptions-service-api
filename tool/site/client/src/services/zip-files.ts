import JSZip from "jszip"
import FileSaver from "file-saver"

interface ZipFileItem {
  fileName: string
  data: string
}

export const zip = async (zipFileName: string, itemsToBeZipped: Array<ZipFileItem>): Promise<void> => {
  const zip = JSZip()

  itemsToBeZipped.forEach(item => {
    zip.file(item.fileName, item.data)
  })

  zip.generateAsync({type: "blob"}).then(zipFile => {
    return FileSaver.saveAs(zipFile, `${zipFileName}.zip`)
  })
}
