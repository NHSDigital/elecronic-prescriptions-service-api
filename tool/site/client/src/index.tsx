import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
