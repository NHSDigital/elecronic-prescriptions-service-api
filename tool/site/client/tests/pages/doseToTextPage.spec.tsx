import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDevSandbox} from "../../src/services/environment"
import DoseToTextPage from "../../src/pages/doseToTextPage"
// import {readBundleFromFile} from "../messages"

const baseUrl = "baseUrl/"
const context: AppContextValue = { baseUrl, environment: internalDevSandbox }

// const doseToTextUrl = `${baseUrl}dose-to-text`
//
// const prescriptionExample = readBundleFromFile("prescriptionOrder.json")

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays dose to text form on render", async () => {
  const container = await renderPage()
  expect(screen.getByText("Dose to Text")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

// test("Displays dose to text result when form data is set", async () => {
//   moxios.stubRequest(doseToTextUrl, {
//     status: 200,
//     response: {
//       results: [{
//         name: "0",
//         success: true
//       }],
//       success: true,
//       request: "JSON Request",
//       response: "JSON Response"
//     }
//   })

// })

async function renderPage() {
  const {container} = renderWithContext(<DoseToTextPage />, context)
  await waitFor(() => screen.getByText("Dose to Text"))
  return container
}
