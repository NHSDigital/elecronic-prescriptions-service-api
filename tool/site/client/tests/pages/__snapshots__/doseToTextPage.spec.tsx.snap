// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays dose to text form on render 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Dose to Text</label></h1>
<p>This form only accepts valid FHIR JSON of the type: prescription-order Bundles OR MedicationRequest</p>
<p>A valid example Bundle can be found<a href=\\"https://simplifier.net/guide/NHSDigital-Medicines/Home/Examples/AllExamples/Messageprescription-order/
        PrescriptionOrderHomecarePrepare.guide.md?version=current\\"> Here</a>.</p>
<p>The dosageInstruction field can be edited independently of the rest of the message for testing purposes.</p>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <div class=\\"nhsuk-form-group\\"><textarea class=\\"nhsuk-textarea\\" name=\\"doseToTextRequest\\" id=\\"doseToTextRequest\\" rows=\\"20\\"></textarea></div>
    </fieldset>
  </div>
  <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Convert</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>
</form>"
`;

exports[`Displays dose to text result 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\"> Dose to Text Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<div class=\\"nhsuk-table-responsive\\">
  <table class=\\"nhsuk-table\\">
    <thead class=\\"nhsuk-table__head\\">
      <tr class=\\"nhsuk-table__row\\">
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Medication Identifier</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Dosage translation</th>
      </tr>
    </thead>
    <tbody class=\\"nhsuk-table__body\\">
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">a54219b8-f741-4c47-b662-e4f8dfa49ab6</td>
        <td class=\\"nhsuk-table__cell\\">Inject 5 times a day - Subcutaneous route - for 10 days</td>
      </tr>
    </tbody>
  </table>
</div>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22Request%20(FHIR)%22\\" download=\\"message\\">Download</a></div><pre class=\\"sc-eCImPb WCiLo\\">\\"Request (FHIR)\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22Response%20(FHIR)%22\\" download=\\"message\\">Download</a></div><pre class=\\"sc-eCImPb WCiLo\\">\\"Response (FHIR)\\"</pre>
    </div>
  </details>
</div>
<div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>"
`;
